docs/reference/remote-authenticator.md

```diff
@@ -0,0 +1,143 @@
+nav_order: 62
+has_children: false
+---
+
+# Remote Authenticator
+
+Remote Authenticator is a pluggable architecture for lakeFS which allows you to integrate any existing organizational identity policies and infrastructure into the  authentication mechanism of lakeFS. This flexibility will enable you to fulfill any security requirement without allowing lakeFS direct access to your identity infrastructure.
+
+## Architecture
+
+The lakeFS authentication flow is detailed the diagram below.
+
+```mermaid
+sequenceDiagram
+    participant A as lakeFS Client
+    participant B as lakeFS Server
+    participant C as Remote Authenticator
+    participant D as IdP
+    A->>B: Submit login form
+    B->>C: POST user credentials
+    C->>D: IdP request
+    D->>C: IdP response
+    C->>B: Auth response
+    B->>A: auth JWT
+```
+
+The Remote Authenticator's job is to abstract away the complexities of existing infrastructure and implement a standard interface, which lakeFS can use to resolve user identity and manage access to lakeFS. This loose coupling allows you to implement Remote Authenticator to work with any existing identity backend without providing lakeFS with direct access to your identity infrastructure.
+
+## The Interface
+
+To allow lakeFS to work with a Remote Authenticator, you'll first need to add the following YAML to your lakeFS configuration
+
+```yaml
+auth:
+    remote_authenticator:
+        enabled: true
+        endpoint: <url-to-remote-authenticator-endpoint>
+        default_user_group: "Developers"
+    ui_config:
+        logout_url: /logout
+        login_cookie_names:
+            - internal_auth_session
+```
+
+- `auth.remote_authenticator.enabled` - set lakeFS to use the remote authenticator
+- `auth.remote_authenticator.endpoint` - an endpoint where the remote authenticator is able to receive a POST request from lakeFS
+- `auth.remote_authenticator.default_user_group` - the group assigned by default to new users
+- `auth.ui_config.logout_url` - the URL to redirect the browser when clicking the logout link in the user menu
+- `auth.ui_config.login_cookie_names` - the name of the cookie(s) lakeFS will set following a successful authentication. The value is the authenticated user's JWT
+
+A Remote Authenticator implementation should expose a single endpoint, which receives the following JSON request:
+
+```JSON
+{
+    "username": "testy.mctestface@example.com",
+    "password": "Password1"
+}
+```
+
+and returns a JSON response like this:
+
+```JSON
+{
+    "external_user_identifier": "TestyMcTestface"
+}
+```
+
+If the Remote Authenticator returns any HTTP status in the 2xx range, lakeFS considers this a successful authentication. Any HTTP status < 200 or > 300 is considered a failed authentication. If the Remote Authenticator returns a non-empty value for the `external_user_identifier` property along with a success HTTP status, lakeFS will show this identifier instead of an internal lakeFS user identifier in the UI.
```

Comment:
I would recommend restructuring this to be more clear around the response details, which break down into two: 

1. The HTTP response 
2. The optional `external_user_identifier` value

At the moment it's a bit back & forth. Perhaps put the example JSON after the recommendation, and also show it as an HTTP response with the code too. You could show example responses of a successful and unsuccessful auth.

Reply:
Added example req and un/successful resp examples